### PR TITLE
macOS: set activation policy by default if app does not have a bundle identifier

### PR DESCRIPTION
**Pull request by nicoburns**
- [x] Tested on all platforms changed
- [ ] Added an entry to the `changelog` module if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

This is an alternative fix for https://github.com/rust-windowing/winit/issues/3958 which makes the fix within winit itself rather than in the `window` example like https://github.com/rust-windowing/winit/pull/3959. I think something like this is probably the better fix as it means that users don't all have to set the activation policy manually if they want keyboard support.

### CODE REVIEW

File: src/platform_impl/apple/appkit/app_state.rs [hunk 1]
```diff
@@ -109,6 +109,8 @@ impl AppState {
         trace_scope!("NSApplicationDidFinishLaunchingNotification");
         self.is_launched.set(true);
 
+        
+
```
Comment (kchibisov) [inline src/platform_impl/apple/appkit/app_state.rs:113]:
```suggestion
```

**Comment by purajit** [comment]
(As the author of https://github.com/rust-windowing/winit/pull/3920 that caused this) Thanks for doing this so quickly! Makes sense to me.

File: src/platform_impl/apple/appkit/app_state.rs [hunk 2]
```diff
@@ -117,6 +117,12 @@ impl AppState {
         // to allow the package manifest to define behavior via LSUIElement.
         if self.activation_policy.is_some() {
             app.setActivationPolicy(self.activation_policy.unwrap());
+        } else {
+            let is_bundle =
+                unsafe { NSRunningApplication::currentApplication().bundleIdentifier().is_some() };
+            if !is_bundle {
```
Comment (madsmtm) [inline src/platform_impl/apple/appkit/app_state.rs:123]: Please add a comment here, perhaps linking to https://github.com/rust-windowing/winit/issues/3958

**Review by madsmtm — APPROVED** [review]
Ah, the difference between bundled and unbundled apps continue to be my bane!

**Pull request merged**